server.py

```diff
@@ -99,34 +102,54 @@ def handle_play(url:str,loop:bool):
 
 def download_next_video_in_list(playlist, current_index):
     next_index = current_index + 1 
-    if next_index < len(playlist):
-        video_url = playlist[next_index]
-        if video_cache.find(Cache.get_video_id(video_url)) is None:
-            video_cache.add(video_url)
+    if next_index==(len(playlist)):
+        next_index = 0
+    video_url = playlist[next_index]
+    if video_cache.find(Cache.get_video_id(video_url)) is None:
+        video_cache.add(video_url)
 
+# return the result of create_ffmpeg_stream
 def download_and_play_video(url,loop):
     video_path = video_cache.find(Cache.get_video_id(url))
     if video_path is None:
         video_cache.add(url)
         video_path = video_cache.find(Cache.get_video_id(url))
-    create_ffmpeg_stream(video_path, State.PLAYING,loop)
+    return create_ffmpeg_stream(video_path, State.PLAYING,loop)
 
-def handle_playlist(playlist_url:str):
+def handle_playlist(playlist_url:str, loop:bool):
     playlist = Playlist(playlist_url)
     # Update process state
     if State.INTERLUDE in process_dict:
         stop_video_by_type(State.INTERLUDE)
     # Stop interlude
-    for i in range(len(playlist)):
-        video_url = playlist[i]
-        video = YouTube(video_url)
-        # Only play age-unrestricted videos to avoid exceptions
-        if not video.age_restricted:
-            current_video_dict["title"] = video.title
-            current_video_dict["thumbnail"] = video.thumbnail_url 
-            # Start downloading next video
-            threading.Thread(target=download_next_video_in_list, args=(playlist, i),).start()
-            download_and_play_video(video_url)
+    if loop:
```

Comment:
how about we do something like
```py
while True:
  for i in range(len(playlist)):
    # insert usual playlist logic here for iterating
    pass
  if not loop:
    break
```


then we dont need the if/else, the single while and nested for loop will account for both looping and single time playlist

Reply:
done